tests/tests.js

```diff
@@ -257,7 +257,8 @@ var runTests = function runTests(arrayFrom) {
 		});
 
 		var hasMap = 'Map' in global;
-		t.test('works with Map objects', { 'skip': !hasMap }, function (te) {
+		var hasMapIterator = hasMap && ((hasSymbols && Map.prototype[Symbol.iterator]) || Map.prototype['@@iterator'] || Map.prototype.iterator);
```

Comment:
We need checks so that in every version of a browser that ships Map and/or Set, we use the proper way to iterate it, and never use a forgeable way to iterate it.